feat(vite): add v8 coverage support to vitest generator

mpgregson opened the pull request:
## Current Behavior
Vitest generator has no option for v8 coverage and defaults to c8

## Expected Behavior
Vitest generator should allow (and default to) v8 coverage 

## Related Issue(s)
Fixes #17886

mpgregson commented:
I've been unable to verify this is working as expected as I received errors when trying to create a new workspace using the locally published package (errors seemed unrelated to my changes).

Could someone help me test/verify that it's working as expected?

barbados-clemens commented on packages/vite/migrations.json:
```diff
@@ -75,6 +75,10 @@
         "@vitest/coverage-istanbul": {
           "version": "^0.32.0",
           "alwaysAddToPackageJson": false
+        },
+        "@vitest/coverage-v8": {
+          "version": "^0.32.0",
+          "alwaysAddToPackageJson": false
```
no change to the migration file is needed here. these are for past versions of nx.

mpgregson replied:
I've removed all changes to this file (assuming that's what you meant?)

barbados-clemens commented on packages/vite/src/utils/versions.ts:
```diff
@@ -11,5 +11,6 @@ export const happyDomVersion = '~9.20.3';
 // Coverage providers
-export const vitestCoverageC8Version = '~0.32.0';
-export const vitestCoverageIstanbulVersion = '~0.32.0';
+export const vitestCoverageC8Version = '^0.32.0';
+export const vitestCoverageV8Version = '^0.32.0';
+export const vitestCoverageIstanbulVersion = '^0.32.0';
```
let's leave these as patch only to prevent downstream issues with other frameworks. 

ref: https://github.com/nrwl/nx/pull/18010

mpgregson replied:
Updated. Hopefully I've understood correctly

barbados-clemens reviewed (changes requested):
Just a couple changes and you'll want to rebase with master.

mpgregson commented:
> Just a couple changes and you'll want to rebase with master.

I think I've addressed the comments correctly, and rebased. Let me know if any further changes are required 👍
